src/auth/impersonated.ts

```diff
@@ -126,6 +127,35 @@ export class Impersonated extends OAuth2Client implements IdTokenProvider {
     this.endpoint = options.endpoint ?? 'https://iamcredentials.googleapis.com';
   }
 
+  /**
+   * Signs some bytes.
+   *
+   * {@link https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/signBlob Reference Documentation}
+   * @param blobToSign String to sign.
+   * @return <SignBlobResponse> denoting the keyyID and signedBlob in base64 string
+   */
+
+  async sign(blobToSign: string): Promise<SignBlobResponse> {
+    await this.sourceClient.getAccessToken();
+    const name = `projects/-/serviceAccounts/${this.targetPrincipal}`;
+    const u = `${this.endpoint}/v1/${name}:signBlob`;
+    const body = {
+      delegates: this.delegates,
+      payload: Buffer.from(blobToSign).toString('base64'),
+    };
+    const res = await this.sourceClient.request<SignBlobResponse>({
+      url: u,
+      data: body,
+      method: 'POST',
+    });
+    return res.data;
+  }
+
+  /** The service account email to be impersonated. */
+  getServiceAccountEmail(): string {
+    return this.targetPrincipal;
+  }
```

Comment:
[Optional] I think target principal would be clearer in this case.

```suggestion
  getTargetPrincipal(): string {
    return this.targetPrincipal;
  }
```